packages/core/src/js/feedback/FeedbackForm.tsx

```diff
@@ -0,0 +1,143 @@
+import { captureFeedback, lastEventId } from '@sentry/core';
+import type { SendFeedbackParams } from '@sentry/types';
+import * as React from 'react';
+import type { KeyboardTypeOptions } from 'react-native';
+import { Alert, Text, TextInput, TouchableOpacity, View } from 'react-native';
+
+import { defaultConfiguration } from './defaults';
+import defaultStyles from './FeedbackForm.styles';
+import type { FeedbackFormProps, FeedbackFormState, FeedbackFormStyles,FeedbackGeneralConfiguration, FeedbackTextConfiguration } from './FeedbackForm.types';
+
+/**
+ * @beta
+ * Implements a feedback form screen that sends feedback to Sentry using Sentry.captureFeedback.
+ */
+export class FeedbackForm extends React.Component<FeedbackFormProps, FeedbackFormState> {
+  public constructor(props: FeedbackFormProps) {
+    super(props);
+
+    const config: FeedbackGeneralConfiguration = { ...defaultConfiguration, ...props };
+    this.state = {
+      isVisible: true,
+      name: config.useSentryUser.name,
+      email: config.useSentryUser.email,
+      description: '',
+    };
+  }
+
+  public handleFeedbackSubmit: () => void = () => {
+    const { name, email, description } = this.state;
+    const { onFormClose } = { ...defaultConfiguration, ...this.props };
+    const config: FeedbackGeneralConfiguration = { ...defaultConfiguration, ...this.props };
+    const text: FeedbackTextConfiguration = { ...defaultConfiguration, ...this.props };
+
+    const trimmedName = name?.trim();
+    const trimmedEmail = email?.trim();
+    const trimmedDescription = description?.trim();
+
+    if ((config.isNameRequired && !trimmedName) || (config.isEmailRequired && !trimmedEmail) || !trimmedDescription) {
+      Alert.alert(text.errorTitle, text.formError);
+      return;
+    }
+
+    if ((config.isEmailRequired || trimmedEmail.length > 0) && !this._isValidEmail(trimmedEmail)) {
```

Comment:
Since we are using custom regex for the email validation, let's add an option to enable/disable the validation.

Reply:
Added an option with c45a5e65835f7aeed05bbd12a5747294b0c121c8